### ISSUE TITLE
Naming a constructor `PS` breaks the javascript

### ISSUE THREAD

**Issue by chrismshelton** [purescript/purescript#3505]
Sorry if this has already been reported, I searched but I couldn't find anything.

Naming a constructor `PS` breaks the resulting javascript:

```purescript
module Main where                                                                                                                                                                                           
                                                                                                                                                                                                            
import Prelude                                                                                                                                                                                              
import Effect (Effect)                                                                                                                                                                                      
import Effect.Console (log)                                                                                                                                                                                 
                                                                                                                                                                                                            
data P = PS                                                                                                                                                                                                 
                                                                                                                                                                                                            
main :: Effect Unit                                                                                                                                                                                         
main = do                                                                                                                                                                                                   
  log "Hello sailor!"
```

When compiled with `pulp build --to output/all.js`, and loaded onto a page, I get the error:

```
Uncaught TypeError: Cannot read property 'Effect' of undefined
    at PS.Main (all.js:25)
    at all.js:38
```

The produced javascript is:

```
// Generated by purs bundle 0.12.0
var PS = {};
(function(exports) {
    "use strict";

  exports.log = function (s) {
    return function () {
      console.log(s);
      return {};
    };
  };
})(PS["Effect.Console"] = PS["Effect.Console"] || {});
(function(exports) {
  // Generated by purs version 0.12.0
  "use strict";
  var $foreign = PS["Effect.Console"];
  var Data_Show = PS["Data.Show"];
  var Data_Unit = PS["Data.Unit"];
  var Effect = PS["Effect"];
  exports["log"] = $foreign.log;
})(PS["Effect.Console"] = PS["Effect.Console"] || {});
(function(exports) {
  // Generated by purs version 0.12.0
  "use strict";
  var Effect = PS["Effect"];
  var Effect_Console = PS["Effect.Console"];
  var Prelude = PS["Prelude"];                 
  var PS = (function () {
      function PS() {

      };
      PS.value = new PS();
      return PS;
  })();
  var main = Effect_Console.log("Hello sailor!");
  exports["PS"] = PS;
  exports["main"] = main;
})(PS["Main"] = PS["Main"] || {});
PS["Main"].main();
```

I am somewhat surprised by this, but can confirm the following also breaks:

```javascript
      A = 123;                                                                                                                                                                                              
                                                                                                                                                                                                            
      var fn = function() {                                                                                                                                                                                 
        console.log (A);                                                                                                                                                                                    
        var A = 234;
      };                                                                                                                                                                                                    
                                                                                                                                                                                                            
      fn();
```

logs "undefined" instead of "123". So it seems the later presence of a variable "PS" inside a function causes the global PS to not be seen, even though the local one hasn't been defined yet.

**Comment by natefaubion** [comment]
This is a product of `purs bundle`. It puts everything in a global namespace which is configured with the `--namespace` flag. You can configure a different namespace to avoid a clash. This is arguable a bug in `purs bundle` which code generate code that does not result in a clash by passing in the namespace object as an argument.

**Comment by mhcurylo** [comment]
I will take a look.

**Comment by hdgarrood** [comment]
This was fixed in #3533
